lib/tasks/generate-from-blueprint.js

```diff
@@ -29,18 +29,51 @@ class GenerateTask extends Task {
     let name = options.args[0];
     let noAddonBlueprint = ['mixin', 'blueprint-test'];
 
-    let mainBlueprint = this.lookupBlueprint(name, options.ignoreMissingMain);
-    let testBlueprint = this.lookupBlueprint(`${name}-test`, true);
+    let entity = {
+      name: options.args[1],
+      options: parseOptions(options.args.slice(2)),
+    };
 
-    // lookup custom addon blueprint
-    let addonBlueprint = this.lookupBlueprint(`${name}-addon`, true);
+    let baseBlueprintOptions = {
+      target: this.project.root,
+      entity,
+      ui: this.ui,
+      analytics: this.analytics,
+      project: this.project,
+      settings: this.settings,
+      testing: this.testing,
+      taskOptions: options,
+      originBlueprintName: name,
+    };
+
+    let mainBlueprintOptions = merge({}, baseBlueprintOptions, options);
+    let testBlueprintOptions = merge({}, mainBlueprintOptions, { installingTest: true });
+    let addonBlueprintOptions = merge({}, mainBlueprintOptions, { installingAddon: true });
```

Comment:
Let's use either `Object.assign` or `{ ... }` here, I don't think we need to use `merge` from lodash

Reply:
Yup, updated!

I assume we can get rid of `merge` in other places as well?

Reply:
Yep!